### DISCUSSION TITLE
Emgu CV reduced Android support

### DISCUSSION THREAD

**Discussion by user20112** [emgucv/emgucv#432] [General]
Good afternoon,
I have been supporting a very old android version and recently updated to Emgu 4. I noticed that the minimum android version has changed to drop android 6 support ? is there any log of why this was required?

Thanks for your time I hope your day is going well

**Comment by emgucv** [comment]
FYI, 

Google Play store has required target API Level of Android 10 as of Aug 2020 for new apps and Nov 2020 for app updates. 

See here:
https://support.google.com/googleplay/android-developer/answer/9859152?visit_id=637497222519901576-2267939769&rd=1#targetsdk&zippy=%2Capk-version-requirements-for-the-play-console%2Ctarget-api-level-requirements-for-the-play-console

Open CV's NDK toolchain target API level of 21 (Android 5). Our native android library is compiled with Open CV toolchain. In theory you can set target level as small as API 21:
https://github.com/opencv/opencv/blob/master/platforms/android/android.toolchain.cmake#L246

Our demo project set minSDKVersion=24, which is Android 7 (Release in 2016). As far as I can remember, this is due to the use of the API from Android 10 to request App permission for our demo, which is not available on Android 6 and will results in crashes.
